spring-cloud-starter-stream-source-load-generator/src/main/resources/application.yml

```diff
@@ -0,0 +1,6 @@
+spring:
+  cloud:
+    stream:
+      bindings:
+        output:
+          contentType: application/x-java-serialized-object
```

Comment:
1. Blank line in the end of each file.
2. We can't enforce `application.yml(properties)` on the library level.

You may consider to do `EnvironmentPostProcessor` (https://github.com/spring-cloud-stream-app-starters/aggregator/blob/master/spring-cloud-starter-stream-processor-aggregator/src/main/java/org/springframework/cloud/stream/app/aggregator/processor/ExcludeStoresAutoConfigurationEnvironmentPostProcessor.java#L36) or somewhere on the programming level. But still: what is the purpose to do this strongly? Can't we somehow make this project JSON-compatible to avoid any reconfiguration and allow end-user to do his/her own decisions about content-type? 

Thanks

Reply:
* new line added
* are you seeing specific compilation warnings? doing `./mvnw clean install -PgenerateApps` from the top level directory, i don't see any warning indicators
* in regards to application.yml, i was under the impression this is where properties go in boot - can you elaborate on why they can't be enforced? 
* in regards to JSON compatible, i think this goes back to the same discussion we had in another PR with @cppwfs in terms of task-launcher, i had just made the code work with what it was expecting here. if we make one component JSON-compatible it would seem like we would want symmetry across the board which sounds like it would need more discussion (?)

Reply:
Thanks for new lines. That's not critical, indeed, but still it is annoying during review and merge. You can configure your IDE to do that for you automatically on file save.

About warnings: I was just under impression that there are enough deprecation in Boot 2.0, so, some tests might be effected like in your PR for `rabbit`.

Regarding the `application.yml`. Well, we don't develop here the target, finished Spring Boot application. That is end-user responsibility to manage such a configuration externally. We do here just a *starter*. The target self-contained application might be out of our hands. More over we may confuse the scanning system (as well as end-users) with the extra `application.yml` packaged in one or more jars in the classpath.  I really don't have too much arguments now do not do that, but I would stay away of some experiment what will happen in the target application (or SCDF ?) if we package `application.yml` into the jar.

I think what you are going to fix with that `application.yml` is out of `main` code responsibility. That is just a test-case failing. So, and that one we should fix so far. Everything rest regarding JSON-compatibility should be addressed in the separate issue, in the appropriate project. Let's don't mix concerns on the discrete microservice level!

Reply:
okay - not too sure what "the discreet microservice level" means, but my main goals when starting to make an attempt at fixing these repos was:

1 - make the project compile and tests pass with the types they currently expected where possible (it would feel odd to deserialize JSON or work with it directly in test cases, if a method returns MyType i expect to get MyType)

2 - test produced artifacts making sure they are usable in data flow

i haven't done #2 for each PR as it would be quite a bit time consuming and I'm relying on the individual project tests/patterns, but #1 has been part of all.

so with that said, and correct me if I am wrong -- i think what you are saying is "don't force a contentType in starter application.yml files but forcing conversion to strict type in test properties is fine as thats what the test is expecting (or wants to work with)" -- would that be correct? if so then i think i see better where you are going with this -- the JSON default is sent over the wire and left for the client to deal with be it by deserializing how they wish via code or by setting a contentType property to have conversion done for them. 

AFAICT from what I have seen so far, the downside of letting JSON fly as the default and not setting a contentType to return what the app used to (forcing in application.yml) is things break on the data flow side as types are generally expected over JSON currently. but i suppose the "fix" for that is to set a property to define the expected content type is by the client (i.e.: in DF shell) and that is the separation between a "stream app" and a "data flow app"?

Reply:
okay - not too sure what "the discreet microservice level" means, but my main goals when starting to make an attempt at fixing these repos was:

1 - make the project compile and tests pass with the types they currently expected where possible (it would feel odd to deserialize JSON or work with it directly in test cases, if a method returns MyType i expect to get MyType)

2 - test produced artifacts making sure they are usable in data flow

i haven't done #2 for each PR as it would be quite a bit time consuming and I'm relying on the individual project tests/patterns, but #1 has been part of all.

so with that said, and correct me if I am wrong -- i think what you are saying is "don't force a contentType in starter application.yml files but forcing conversion to strict type in test properties is fine as thats what the test is expecting (or wants to work with)" -- would that be correct? if so then i think i see better where you are going with this -- the JSON default is sent over the wire and left for the client to deal with be it by deserializing how they wish via code or by setting a contentType property to have conversion done for them. 

AFAICT from what I have seen so far, the downside of letting JSON fly as the default and not setting a contentType to return what the app used to (forcing in application.yml) is things break on the data flow side as types are generally expected over JSON currently. but i suppose the "fix" for that is to set a property to define the expected content type is by the client (i.e.: in DF shell) and that is the separation between a "stream app" and a "data flow app"?

Reply:
ok so i updated the PR to remove the `application.yml` and move the content type setting into the test case otherwise the project wouldn't build. i used `application/octet-stream` since thats what the README indicates it should be and the test passes.

when used in data flow along with `throughput`, for example:

`stream create --name testLoad --definition "load-generator | throughput" --deploy`

the `throughput` sink receives messages properly. i am assuming this works out of the box because `https://github.com/spring-cloud-stream-app-starters/throughput/blob/master/spring-cloud-starter-stream-sink-throughput/src/main/java/org/springframework/cloud/stream/app/throughput/sink/ThroughputSinkConfiguration.java#L84` -- the method signature is `Message<?>` and marked with `@ServiceActivator`. it seems according to https://docs.spring.io/spring-integration/api/org/springframework/integration/annotation/ServiceActivator.html I'm guessing thats why a JSON input is being converted automatically into a `Message` in this case